### PR TITLE
Several optimizations are applied to this version.

### PR DESCRIPTION
**Pull request by pin-yu**
1. Replace FixedThreadPool to WorkStealingPool to reduce handover latency.
2. Append 10000 blocks at once if there is no empty block.
3. Remove some profiling code.

### CODE REVIEW

File: src/main/java/org/vanilladb/core/storage/tx/concurrency/ConcurrencyMgr.java [hunk 1]
```diff
@@ -178,4 +179,8 @@ public void lockRecordFileHeader(BlockId blk) {
 	public void releaseRecordFileHeader(BlockId blk) {
 		lockTbl.release(blk, txNum, LockTable.X_LOCK);
 	}
+	
+	public ReentrantLock getLockForFileHeader(BlockId blk) {
+		return null;
```
Comment (SLMT) [inline src/main/java/org/vanilladb/core/storage/tx/concurrency/ConcurrencyMgr.java:184]: We better give it a global latch here to ensure that VanillaCore still works alone.

File: src/main/java/org/vanilladb/core/storage/record/RecordFile.java [hunk 2]
```diff
@@ -398,15 +413,8 @@ private void appendBlock() {
...
-		if (!isTempTable()) {
-			fhpWaitCount.incrementAndGet();
-			tx.concurrencyMgr().lockRecordFileHeader(headerBlk);
-			fhpReleaseCount.incrementAndGet();
-			fhpWaitCount.decrementAndGet();
-
-		}
-		return new FileHeaderPage(fileName, tx);
+		FileHeaderPage fhp = new FileHeaderPage(fileName, tx);
+		return fhp;
```
Comment (SLMT) [inline src/main/java/org/vanilladb/core/storage/record/RecordFile.java:417]: It seems like there is no corresponding modification for `delete()` and `insert(RecordId)`. We better add a warning like `TODO` or `XXX` here.

File: src/main/java/org/vanilladb/core/storage/file/io/jaydio/JaydioDirectIoChannel.java [hunk 3]
```diff
@@ -57,9 +80,11 @@ public int write(IoBuffer buffer, long position) throws IOException {
...
 			int writeSize = fileChannel.write(jaydioBuffer.getAlignedDirectByteBuffer(), position);
 
-			// Check if we need to update the size
-			if (position + writeSize > fileSize)
-				fileSize = position + writeSize;
+			long expectedSize = position + writeSize;
+			// Check if an extra block is needed for this write.
+			if (expectedSize > pretendFileSizes.get(fileName)) {
+				pretendFileSizes.put(fileName, expectedSize);
+			}
```
Comment (SLMT) [inline src/main/java/org/vanilladb/core/storage/file/io/jaydio/JaydioDirectIoChannel.java:87]: Maybe we need to also update `realFileSize`?

**Review by SLMT — CHANGES_REQUESTED** [review]
Please check the comments

File: src/main/java/org/vanilladb/core/storage/tx/concurrency/ConcurrencyMgr.java (hunk 1 above)
Comment (pin-yu) [reply src/main/java/org/vanilladb/core/storage/tx/concurrency/ConcurrencyMgr.java:184]: Got it

File: src/main/java/org/vanilladb/core/storage/record/RecordFile.java (hunk 2 above)
Comment (pin-yu) [reply src/main/java/org/vanilladb/core/storage/record/RecordFile.java:417]: I've modified both to reduce the future works

File: src/main/java/org/vanilladb/core/storage/file/io/jaydio/JaydioDirectIoChannel.java (hunk 3 above)
Comment (pin-yu) [reply src/main/java/org/vanilladb/core/storage/file/io/jaydio/JaydioDirectIoChannel.java:87]: Done! Thanks for the suggestion

**Review by SLMT — APPROVED** [review]
Good to go!

**Pull request merged**